rules/mapping_json_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package rules
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/m3db/m3metrics/policy"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewMappingRuleJSON(t *testing.T) {
+	id := "mr_id"
+	name := "mr_name"
+	fixture := testMappingRuleView(id, name)
+	expected := MappingRuleJSON{
+		ID:                  id,
+		Name:                name,
+		Filter:              "filter",
+		Policies:            []policy.Policy{},
+		CutoverMillis:       0,
+		LastUpdatedBy:       "",
+		LastUpdatedAtMillis: 0,
+	}
+	require.EqualValues(t, expected, NewMappingRuleJSON(fixture))
+}
+
+func TestMappingRuleView(t *testing.T) {
+	id := "id"
+	name := "name"
+	fixture := testMappingRuleJSON(id, name)
+	expected := &MappingRuleView{
+		ID:       id,
+		Name:     name,
+		Filter:   "filter",
+		Policies: []policy.Policy{},
+	}
+	require.EqualValues(t, expected, fixture.MappingRuleView())
+}
+func TestMappingRuleEqual(t *testing.T) {
```

Comment:
`TestMappingRuleJSONEqual` might be better? Conventionally the type name in the test should match the name of the type you are testing, so if in the future we add an `Equal` method for the `mappingRule` type, it does not conflict with this test function.